src/TorchSharp/NN/Module.cs

```diff
@@ -181,18 +183,32 @@ protected void _toEpilog(Device device, ScalarType dtype)
                                 var p = new Parameter(t, param.requires_grad);
                                 field.SetValue(this, p);
                                 ConditionallyRegisterParameter(fieldName, p);
+                                alreadyHandled.Add(p.handle);
                                 break;
                             }
 
                         case Tensor tensor when (device.type != tensor.device_type || device.index != tensor.device_index): {
                                 var t = tensor.to(dtype, device);
                                 field.SetValue(this, t);
                                 ConditionallyRegisterBuffer(fieldName, t);
+                                alreadyHandled.Add(t.handle);
                                 break;
                             }
                         }
                     }
 
+                    foreach (var (name, param) in named_parameters().ToList()) {
```

Comment:
>ToList [](http://example.com/codeflow?start=69&length=6)

why calling ToList()? this apply to other changes in this file.

Reply:
Because the call to "ConditionallyRegister" will modify the collection underlying `named_parameters(),` so I need a clone. It may be faster to call `.Where((n,p) => !alreadyHandled.Contains(p.handle)).ToList()` and get a smaller list out, but I'm always worried about using LINQ in that way, not sure about the performance implications.

Reply:
ok, make sense then.

Reply:
Would ToArray() be faster? I've always assumed ToList() is faster.

Reply:
I didn't measure it myself, but it may be worth measuring it with your scenario. this thread https://stackoverflow.com/questions/1105990/is-it-better-to-call-tolist-or-toarray-in-linq-queries is interesting to read. At the end measuring can give concrete answer.